backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/domain/use_cases/prior_notification/CreateOrUpdateManualPriorNotification.kt

```diff
@@ -174,12 +183,15 @@ class CreateOrUpdateManualPriorNotification(
         note: String?,
         pnoTypes: List<PriorNotificationType>,
         portLocode: String,
+        updatedBy: String?,
         computedVesselFlagCountryCode: CountryCode?,
         computedVesselRiskFactor: Double?,
         isPartOfControlUnitSubscriptions: Boolean,
     ): PNO {
         val allPorts = portRepository.findAll()
 
+        val authorTrigram = existingMessageValue?.authorTrigram
+        val createdBy = existingMessageValue?.createdBy ?: updatedBy
```

Comment:
Il y a un problème de logique non ?

On ne veut pas garder `existingMessageValue?.createdBy` ?

```suggestion
        val createdBy = existingMessageValue?.createdBy
```

Reply:
S'il n'y a pas de PNO manuel existant, `updatedBy` représente en fait le nom du créateur du PNO. Je peux renommer le param en `author` ou `createdOrUpdatedBy` pour clarifier ça peut-être ?

Mais ça révèle une erreur sur le `updatedBy` par contre. Du coup :

```kt
val createdBy = existingMessageValue?.createdBy ?: author
val updatedBy = if (existingMessageValue != null) author else null
```

?